### PR TITLE
RegisterMetricNames does not create new per-day index entries for new dates for existing timeseries

### PR DESCRIPTION
**Pull request by rtm0**
### Describe Your Changes

Once the timeseries is in tsidCache, new entries won't be created in per-day index because the RegisterMetricNames() code does consider different dates for the same timeseries. So this case has been added.

The same bug exists for AddRows() but it is not manifested because the index entries are finally created in updatePerDateData().

RegisterMetricNames also updated to increase the newTimeseriesCreated counter because it actually creates new time series in index.

A unit tests has been added that check all possible data patterns (different metric names and dates) and code branches in both RegisterMetricNames and AddRows. The total number of new unit tests is around 100 which increaded the running time of storage tests by 50%. The PR #6871 should help here.

### Checklist

The following checks are **mandatory**:

- [x] My change adheres [VictoriaMetrics contributing guidelines](https://docs.victoriametrics.com/contributing/).

### CODE REVIEW

**Comment by hagen1778** [comment]
>Once the timeseries is in tsidCache, new entries won't be created in per-day index because the RegisterMetricNames() code does consider different dates for the same timeseries. So this case has been added.

@rtm0 do I understand it right that this bug is related only to Graphite https://graphite.readthedocs.io/en/stable/tags.html#adding-series-to-the-tagdb handler?

**Comment by rtm0** [comment]
> @rtm0 do I understand it right that this bug is related only to Graphite https://graphite.readthedocs.io/en/stable/tags.html#adding-series-to-the-tagdb handler?

@hagen1778, RegisterMetricNames() is currently used for Graphite only. So yes, you are right. I understand that it is a very narrow use case and the fix is of low importance. But this code is in the codebase and therefore needs to be maintained. Also it turns out that this code can be used for restoring per-day index after it was disabled and then enabled again (something I've been working on and is almost ready, I will send a pr once a few other pending ones a merged in)

 I'd also say that the unit test in this pr is even more important since it covers many different data patterns and code branches in both RegisterMetricNames() and AddRows(), since it allows to intoduce changes to these functions safely.

**Review by hagen1778 — CHANGES_REQUESTED** [review]
Looks good! Thanks for adding well described tests!

Could you please add a note to the [CHANGELOG](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/changelog/CHANGELOG.md#tip) with description of the bugfix? See other bugfixes from contributors for examples.

**Comment by rtm0** [comment]
> Looks good! Thanks for adding well described tests!

Thank you!

> Could you please add a note to the [CHANGELOG](https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/docs/changelog/CHANGELOG.md#tip) with description of the bugfix? See other bugfixes from contributors for examples.

Done.

File: docs/changelog/CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +49,7 @@ The value of `instance` label for those scrape targets will be changed from `<ad
...
 * BUGFIX: [vmalert](https://docs.victoriametrics.com/vmalert): reduce memory usage when parsing responses with big number of metrics in response. The memory usage was increased in [v1.102.0-rc1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases/tag/v1.102.0-rc1) after attempt to reduce CPU usage for heavy loaded vmalerts.
 * BUGFIX: all VictoriaMetrics components: forcefully set owner/group for release tars to 1000:1000. This helps to avoid unpacking [issues](https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6788) on systems with limitations around UID:GID configuration. See [this pull request](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6846).
+* BUGFIX: [Single-node VictoriaMetrics](https://docs.victoriametrics.com/) and `vmstorage` in [VictoriaMetrics cluster](https://docs.victoriametrics.com/cluster-victoriametrics/): Fixes registering metric names in the per-day index for new dates for existing timeseries. See [this](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6872/) for details.
```
Comment (hagen1778) [inline docs/changelog/CHANGELOG.md:52]:
```suggestion
* BUGFIX: [Single-node VictoriaMetrics](https://docs.victoriametrics.com/) and `vmstorage` in [VictoriaMetrics cluster](https://docs.victoriametrics.com/cluster-victoriametrics/): fix metric names registering in the per-day index for new dates for existing time series when making calls to `/tags/tagSeries` and `/tags/tagMultiSeries` handlers of [Grpahite API](https://docs.victoriametrics.com/#graphite-api-usage). See [this](https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6872/) for details.
```

**Review by hagen1778 — APPROVED** [review]
LGTM

**Comment by rtm0** [comment]
Accepted the suggestion. Thanks!

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**